tests/api.py

```diff
@@ -151,27 +151,30 @@ def api_ctx_ownership():
     del ctx2
 
 
-def api_ctx_ownership_subgraph():
-    for shared in (True, False):
-        ctx = ngl.Context()
-        ctx2 = ngl.Context()
-        ret = ctx.configure(ngl.Config(offscreen=True, width=16, height=16, backend=_backend))
-        assert ret == 0
-        ret = ctx2.configure(ngl.Config(offscreen=True, width=16, height=16, backend=_backend))
-        assert ret == 0
-        quad = ngl.Quad()
-        render1 = _get_scene(quad)
-        if not shared:
-            quad = ngl.Quad()
-        render2 = _get_scene(quad)
-        root = ngl.Group([render1.root, render2.root])
-        scene = ngl.Scene.from_params(root)
-        assert ctx.set_scene(render2) == 0
-        assert ctx.draw(0) == 0
-        assert ctx2.set_scene(scene) != 0
-        assert ctx2.draw(0) == 0
-        del ctx
-        del ctx2
+def api_scene_lifetime():
+    """Test if the context is still working properly when we release the user scene ownership"""
+    scene = _get_scene()
+    ctx = ngl.Context()
+    ret = ctx.configure(ngl.Config(offscreen=True, width=16, height=16, backend=_backend))
+    assert ret == 0
+    assert ctx.set_scene(scene) == 0
+    del scene
+    assert ctx.draw(0) == 0
+    del ctx
+
+
+def api_scene_ownership():
+    """Test if part of a graph is shared between 2 different scenes"""
+    shared_geometry = ngl.Quad()
+    scene0 = ngl.Scene.from_params(ngl.RenderColor(geometry=shared_geometry))
+    try:
+        scene1 = ngl.Scene.from_params(ngl.RenderColor(geometry=shared_geometry))
+    except Exception:
+        pass
+    else:
+        del scene1
+        assert False
+    del scene0
```

Comment:
While at it, can you also test with a ngl context attached (in various scenarios) ?
- sharing the graph
- sharing a subpart of the graph
- before and after detaching the initial ngl ctx / scene

Reply:
Sharing a subpart of the graph is already tested at the scene level, so it's not possible at the context level, and it also implies sharing the whole graph.

I'm not sure what you mean by before and after detaching but I added a test to transfer a scene from one context to another.

Reply:
To clarify a bit: if a scene/graph is associated with a rendering context (and is up and running), we want to make sure (and I mean that we need a test for this scenario) that any intervention from the user with the graph or the subgraph won't mess up the rendering context.

Reply:
Added a dedicated commit: `nodes: prevent graph structure changes after scene association`

Reply:
Ok but, what i meant is more about, you associate a graph with a rendering context, you test that draw succeed. Then, you try to ref the first graph from another graph, try to associate it with a scene, then a rendering context if the previous operation succeeded (i believe it should not so skip this step) then you check again that the first rendering context is still working by checking that a draw succeeded. The goal is to ensure that the error path in the attach/init functions does not mess with the first graph.

Reply:
Added `api_scene_resilience` test, but note that I couldn't work with the original scene object directly due to limitation in the Python binding (`.from_params()` is a class constructor and can't be re-used on an existing instance)

Reply:
Ok. Thanks.